utils/terrascript_client.py

```diff
@@ -306,6 +310,66 @@ def populate_additional_providers(self, accounts):
                            alias=alias,
                            assume_role={'role_arn': assume_role})
 
+    def populate_route53(self, desired_state):
+        for zone in desired_state:
+            # provider = zone['provider']
+            account = zone['account']
+            account_name = account['name']
+            origin = zone['origin']
+            records = zone['records']
+
+            # terraform identifier for zone
+            # We need to replace some characters that are not allowed
+            zone_id = origin
+            zone_id = zone_id.replace(".", "__")
+
+            values = {
+                'name': origin,
+            }
+
+            zone_resource = aws_route53_zone(zone_id, **values)
+            self.add_resource(account_name, zone_resource)
+
+            counts = defaultdict(int)
+            for record in records:
+                counts[record['name']] += 1
```

Comment:
Nice usage of the defaultdict :)
Have you seen the `collections.Counter`? Just mentioning, no advantages for this case anyway.

Reply:
Thanks for the pointer! I've dropped the need for `defaultdict` or `collections.Counter` in this case but I'll keep it in mind for future needs!